src/darwin/Framework/CHIP/MTRDevice.h

```diff
@@ -201,6 +241,26 @@ typedef NS_ENUM(NSUInteger, MTRDeviceState) {
                                       completion:(MTRDeviceOpenCommissioningWindowHandler)completion
     MTR_AVAILABLE(ios(17.0), macos(14.0), watchos(10.0), tvos(17.0));
 
+/**
+ * Requests diagnostic logs of the desired types from the device.
+ *
+ * Note: The consumer of this API should move the files that the NSURLs in the MTRDiagnosticLogResult point to
+ * or open it for reading before the completion handler returns. Otherwise, the files will be deleted, and the data will be lost.
```

Comment:
```suggestion
 * or open them for reading before the completion handler returns. Otherwise, the files will be deleted, and the data will be lost.
```